api/src/main/java/org/cardanofoundation/rosetta/api/construction/service/OperationService.java

```diff
@@ -58,33 +56,59 @@ public static List<Operation> convert(TransactionData data,
     return operations;
   }
 
-  private static void fillVoteOperations(TransactionExtraData extraData, List<Operation> operations)
-      throws CborDeserializationException {
-    List<Operation> voteOp = extraData.operations().stream()
-        .filter(o -> o.getType().equals(OperationType.VOTE_REGISTRATION.getValue()))
-        .toList();
-    if (!ObjectUtils.isEmpty(voteOp)) {
-      Operation parsedVoteOperations = ParseConstructionUtil.parseVoteMetadataToOperation(
-          voteOp.getFirst().getOperationIdentifier().getIndex(),
-          extraData.transactionMetadataHex()
-      );
-      operations.add(parsedVoteOperations);
+  public List<String> getSignerFromOperation(NetworkIdentifierType networkIdentifierType,
+      Operation operation) {
+    if (Constants.POOL_OPERATIONS.contains(operation.getType())) {
+      return getPoolSigners(networkIdentifierType, operation);
+    }
+    if (operation.getAccount() != null) {
+      // org.openapitools.client.model.AccountIdentifier.getAddress() is always not null
+      return Collections.singletonList(operation.getAccount().getAddress());
+    }
+    if (operation.getType().equals(OperationType.VOTE_REGISTRATION.getValue())) {
+      return Collections.EMPTY_LIST;
     }
+    validateMetadataForStakingCredential(operation);
```

Comment:
Here I added a validation to avoid NPE. The same validation could be found in the TS version:
https://github.com/cardano-foundation/cardano-rosetta/blob/542053f87109d883f31339194a867c57f065213c/cardano-rosetta-server/src/server/utils/cardano/staking-credentials.ts#L16